### PR TITLE
Bowling results processor

### PR DESCRIPTION
**Pull request by adamvm**
Done with @marcindus

### CODE REVIEW

File: bowling/Files.hpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
```
Comment (WBigaj) [inline bowling/Files.hpp:6]:
```suggestion
#include <vector>

```

File: bowling/Interpreter.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#include "Interpreter.hpp"
```
Comment (WBigaj) [inline bowling/Interpreter.cpp:1]:
```suggestion
#include "Interpreter.hpp"

```

File: bowling/Interpreter.hpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#pragma once
```
Comment (WBigaj) [inline bowling/Interpreter.hpp:1]:
```suggestion
#pragma once

```

File: bowling/Interpreter.hpp [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
```
Comment (WBigaj) [inline bowling/Interpreter.hpp:6]:
```suggestion
#include <vector>

```

File: bowling/bowling.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
+#include "Files.hpp"
+#include "Printer.hpp"
```
Comment (WBigaj) [inline bowling/bowling.cpp:2]:
```suggestion
#include "Printer.hpp"

```

File: bowling/bowling.cpp [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    break;
+  case 3: {
+    processGame(std::string(argv[1]), std::string(argv[2]));
+    break;
+  }
+  default:
+    printHelp();
+  }
+  }
```
Comment (WBigaj) [inline bowling/bowling.cpp]: This switch definitely needs some formatting fixes.

**Review by WBigaj — COMMENTED** [review]
I have looked at your code and there some formatting suggestions.

**Comment by adamvm** [comment]
@ziobron if possible, please review architecture

File: bowling/Bowls.hpp [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  int currentRoll = 0;
+
+public:
+  void roll(int i);
+  int score();
+  [[nodiscard]] bool isSpare(int firstInFrame) const;
+  [[nodiscard]] bool isStrike(int firstInFrame) const;
+  [[nodiscard]] int nextTwoBallsForStrike(int firstInFrame) const;
+  [[nodiscard]] int nextBallForSpare(int firstInFrame) const;
+};
```
Comment (ziobron) [inline bowling/Bowls.hpp:17]: is this class the same as Bowling? Duplication?

File: bowling/Printer.hpp [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+#include <string>
+
+#include "Bowls.hpp"
+#include "Interpreter.hpp"
+
+class Printer {
+public:
+  std::string extractLaneName(const std::string &name);
+  std::stringstream
+  generateSummary(const std::map<std::string, std::string> &results);
```
Comment (ziobron) [inline bowling/Printer.hpp:12]:
```suggestion
  std::stringstream generateSummary(const std::map<std::string, std::string> &results);
```

**Review by ziobron — APPROVED** [review]
For me architecture is fine 👍 
I guess that some methods can be private instead of public.

File: bowling/Bowls.hpp (hunk 7 above)
Comment (adamvm) [reply bowling/Bowls.hpp:17]: Some problem with commit, here is main program file https://github.com/adamvm/bowlingTDD/blob/master/bowling.cpp I will update form my pc

**Comment by adamvm** [comment]
@ziobron will we receive points for this?

**Comment by ziobron** [comment]
I was waiting for the update as you stated above. I'll grant you points later today.

**Pull request closed without merging**